zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -58,87 +125,111 @@ fn cmd_output(cmd: &mut std::process::Command) -> Result<String> {
         output.status.code()
     );
 
-    // Make sure the output is valid UTF-8
-    let s = String::from_utf8(output.stdout)?;
-    Ok(s)
+    // Make sure the output is valid UTF-8 JSON
+    let response = String::from_utf8(output.stdout)?;
+    // zcash-cli returns raw strings without JSON type info.
+    // As a workaround, assume that invalid responses are strings.
+    let response: Value = serde_json::from_str(&response)
+        .unwrap_or_else(|_error| Value::String(response.trim().to_string()));
+
+    Ok(response)
 }
 
 /// Process entry point for `zebra-checkpoints`
+#[tokio::main]
 #[allow(clippy::print_stdout)]
-fn main() -> Result<()> {
+async fn main() -> Result<()> {
     // initialise
     init_tracing();
     color_eyre::install()?;
 
-    // get the current block count
-    let mut cmd = passthrough_cmd();
-    cmd.arg("getblockchaininfo");
+    let args = args::Args::from_args();
 
-    let output = cmd_output(&mut cmd)?;
-    let get_block_chain_info: Value = serde_json::from_str(&output)?;
+    // get the current block count
+    let get_block_chain_info = rpc_output(&args, "getblockchaininfo", None)
+        .await
+        .with_suggestion(|| {
+            "Is the RPC server address and port correct? Is authentication configured correctly?"
+        })?;
 
     // calculate the maximum height
-    let height_limit = block::Height(get_block_chain_info["blocks"].as_u64().unwrap() as u32);
+    let height_limit = get_block_chain_info["blocks"]
+        .try_into_height()
+        .expect("height: unexpected invalid value, missing field, or field type");
 
-    assert!(height_limit <= block::Height::MAX);
     // Checkpoints must be on the main chain, so we skip blocks that are within the
     // Zcash reorg limit.
     let height_limit = height_limit
-        .0
-        .checked_sub(MIN_TRANSPARENT_COINBASE_MATURITY)
-        .map(block::Height)
-        .expect("zcashd has some mature blocks: wait for zcashd to sync more blocks");
-
-    let starting_height = args::Args::from_args().last_checkpoint.map(block::Height);
-    if starting_height.is_some() {
-        // Since we're about to add 1, height needs to be strictly less than the maximum
-        assert!(starting_height.unwrap() < block::Height::MAX);
-    }
+        - HeightDiff::try_from(MIN_TRANSPARENT_COINBASE_MATURITY).expect("constant fits in i32");
+    let height_limit =
+        height_limit.expect("node has some mature blocks: wait for it to sync more blocks");
```

Comment:
Unfortunately, I don't understand what this error message means. The error occurs when the node has less than 100 blocks, which means it's a fresh node. What about something like the following?
```suggestion
        let height_limit = height_limit.expect(
        "The node should have more than {MIN_TRANSPARENT_COINBASE_MATURITY} synced blocks.",
    );
```

Reply:
Thanks for this review, none of these things are blokcking, so I'll fix them in a follow-up PR, to unblock the next PR in this series merging.

Reply:
These are now fixed in https://github.com/ZcashFoundation/zebra/pull/6579/files